docs/content/develop/test.md

```diff
@@ -0,0 +1,220 @@
+   git checkout main && git clean -f . && git checkout -- . && git pull
+   ```
+
+## Add a create test
+
+A create test is a test that creates the target resource and immediately destroys it.
+
+> **Note:** All resources should have a "basic" create test, which uses the smallest possible number of fields. Additional create tests can be used to ensure all fields on the resource are used in at least one test.
+
+{{< tabs "create" >}}
+{{< tab "MMv1" >}}
+1. Using an editor of your choice, create a `*.tf.erb` file in [`mmv1/templates/terraform/examples/`](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples). The name of the file should include the service name, resource name, and a descriptor. For example, `compute_subnetwork_basic.tf.erb`.
+2. Write the Terraform configuration for your test. This should include all of the required dependencies. For example, `google_compute_subnetwork` has a dependency on `google_compute_network`:
+   ```tf
+   resource "google_compute_subnetwork" "primary" {
+     name          = "my-subnet"
+     ip_cidr_range = "10.1.0.0/16"
+     region        = "us-central1"
+     network       = google_compute_network.network.name
+   }
+
+   resource "google_compute_network" "network" {
+     name                    = "my-network"
+     auto_create_subnetworks = false
+   }
+   ```
+3. If beta-only fields are being tested:
+   - Add `provider = google-beta` to every resource in the file.
+4. Modify the configuration to use templated values.
+   - Replace the id of the primary resource you are testing with `<%= ctx[:primary_resource_id] %>`.
+   - Replace fields that are identifiers, like `id` or `name`, with an appropriately named variable. For example, `<%= ctx[:vars]['subnetwork_name'] %>`.
+   - The resulting configuration for the above example would look like this:
+   ```tf
+   resource "google_compute_subnetwork" "<%= ctx[:primary_resource_id] %>" {
+     name          = "<%= ctx[:vars]['subnetwork_name'] %>"
+     ip_cidr_range = "10.1.0.0/16"
+     region        = "us-central1"
+     network       = google_compute_network.network.name
+   }
+
+   resource "google_compute_network" "network" {
+     name                    = "<%= ctx[:vars]['network_name'] %>"
+     auto_create_subnetworks = false
+   }
+   ```
+5. Modify the relevant `RESOURCE_NAME.yaml` file under [magic-modules/mmv1/products](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/products) to include an [`examples`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/provider/terraform/examples.rb) block with your test. The `name` must match the name of your `*.tf.erb` file. For example:
+   ```yaml
+   examples:
+     - !ruby/object:Provider::Terraform::Examples
+       name: "compute_subnetwork_basic"
+       primary_resource_id: "example"
+       vars:
+         subnetwork_name: "example-subnet"
+         network_name: "example-network"
+   ```
+   {{< hint warning >}}
```

Comment:
for some reason this is rendering (for me locally) as

![Screenshot 2023-08-04 at 10 01 36 AM](https://github.com/GoogleCloudPlatform/magic-modules/assets/299979/b1486714-e8aa-4ce7-9c17-501c8bc1d482)

It looks like removing the indentation means that the words get placed correctly inside the colored area - it's maybe not ideal to not have the indentation but possibly the best we can do for now.

Reply:
Yea, it's something that only happens inside the "tab", and I wasn't able to trick it into placing the text where I wanted. I'll just un-indent for now.